contracts/SportMarkets/Rundown/GamesPlayerPropsReceiver.sol

```diff
@@ -0,0 +1,266 @@
+    ITherundownConsumer public consumer;
+
+    mapping(address => bool) public whitelistedAddresses;
+    mapping(uint => mapping(uint8 => bool)) public isValidOptionPerSport;
+    mapping(uint => uint[]) public optionsPerSport;
+
+    address public wrapperAddress;
+
+    /// @notice public initialize proxy method
+    /// @param _owner future owner of a contract
+    function initialize(
+        address _owner,
+        address _consumer,
+        address _playerProps,
+        address[] memory _whitelistAddresses
+    ) public initializer {
+        setOwner(_owner);
+        consumer = ITherundownConsumer(_consumer);
+        playerProps = IGamesPlayerProps(_playerProps);
+
+        for (uint i; i < _whitelistAddresses.length; i++) {
+            whitelistedAddresses[_whitelistAddresses[i]] = true;
+        }
+    }
+
+    /* ========== PLAYER PROPS R. MAIN FUNCTIONS ========== */
+
+    /// @notice receive player props and create markets
+    /// @param _gameIds for which gameids market is created (Boston vs Miami etc.)
+    /// @param _playerIds for which playerids market is created (12345, 678910 etc.)
+    /// @param _options for which options market is created (points, assists, etc.)
+    /// @param _names for which player names market is created (Jimmy Buttler etc.)
+    /// @param _lines number of points assists per option
+    /// @param _linesOdds odds for lines
+    function fulfillPlayerProps(
+        bytes32[] memory _gameIds,
+        bytes32[] memory _playerIds,
+        uint8[] memory _options,
+        string[] memory _names,
+        uint16[] memory _lines,
+        int24[] memory _linesOdds
+    ) external isAddressWhitelisted {
+        for (uint i = 0; i < _gameIds.length; i++) {
+            uint sportId = consumer.sportsIdPerGame(_gameIds[i]);
+            if (isValidOptionPerSport[sportId][_options[i]]) {
+                IGamesPlayerProps.PlayerProps memory player = _castToPlayerProps(
+                    i,
+                    _gameIds[i],
+                    _playerIds[i],
+                    _options[i],
+                    _names[i],
+                    _lines[i],
+                    _linesOdds
+                );
+                // game needs to be fulfilled and market needed to be created
+                if (consumer.gameFulfilledCreated(_gameIds[i]) && consumer.marketPerGameId(_gameIds[i]) != address(0)) {
+                    playerProps.obtainPlayerProps(player, sportId);
+                }
+            }
+        }
+    }
+
+    /// @notice receive resolve properties for markets
+    /// @param _gameIds for which gameids market is resolving (Boston vs Miami etc.)
+    /// @param _playerIds for which playerids market is resolving (12345, 678910 etc.)
+    /// @param _options options (assists, points etc.)
+    /// @param _scores number of points assists etc. which player had
+    /// @param _statuses resolved statuses
+    function fulfillResultOfPlayerProps(
```

Comment:
Doesnt need to be changed now, but perhaps these two methods (fullfilResultOfPlayerProps and fulfilPlayerPropsClResolved) can be merged into one method where the wrapper is just a whitelisted address. Could help with contract size down the line.

Reply:
Yep, I understand. Let it stay for now like this and we can merge those two if/when needed